tests/common/helpers/parallel_utils.py

```diff
@@ -1,5 +1,217 @@
+        with open(self.state_file, 'r+') as f:
+            fcntl.flock(f, fcntl.LOCK_EX)
+            lines = f.readlines()
+            if not lines:
+                raise Exception("State file is empty")
+
+            last_line = lines[-1].strip()
+            _, status_value, acknowledgments, _, _ = last_line.split(',')
+            if status_value != ack_status.value:
+                raise Exception("Cannot acknowledge status {} when status is {}".format(ack_status, status_value))
+
+            f.write("{},{},{},{},{}\n".format(
+                datetime.utcnow(),
+                ack_status.value,
+                int(acknowledgments) + 1,
+                ParallelRole.LEADER.value if is_leader else ParallelRole.FOLLOWER.value,
+                hostname,
+            ))
+
+            f.flush()
+            fcntl.flock(f, fcntl.LOCK_UN)
+
+    def _is_all_acknowledged(self, ack_status: InitialCheckStatus) -> bool:
+        status_value, acknowledgments = self._read_state()
+        return status_value == ack_status.value and acknowledgments >= self.num_followers
+
+    def mark_tests_completed_for_follower(self, hostname: str) -> None:
+        with open(self.state_file, 'r+') as f:
+            fcntl.flock(f, fcntl.LOCK_EX)
+            lines = f.readlines()
+            if not lines:
+                raise Exception("State file is empty")
+
+            last_line = lines[-1].strip()
+            _, status_value, acknowledgments, _, _ = last_line.split(',')
+            f.write("{},{},{},{},{}\n".format(
+                datetime.utcnow(),
+                InitialCheckStatus.TESTS_COMPLETED.value,
+                int(acknowledgments) + 1 if status_value == InitialCheckStatus.TESTS_COMPLETED.value else 1,
+                ParallelRole.FOLLOWER.value,
+                hostname,
+            ))
+
+            f.flush()
+            fcntl.flock(f, fcntl.LOCK_UN)
+
+    def set_new_status(self, new_status: InitialCheckStatus, is_leader: bool, hostname: str) -> None:
+        with open(self.state_file, 'a+') as f:
+            fcntl.flock(f, fcntl.LOCK_EX)
+            f.write("{},{},{},{},{}\n".format(
+                datetime.utcnow(),
+                new_status.value,
+                0,
+                ParallelRole.LEADER.value if is_leader else ParallelRole.FOLLOWER.value,
+                hostname,
+            ))
+
+            f.flush()
+            fcntl.flock(f, fcntl.LOCK_UN)
+
+    def wait_and_acknowledge_status(self, expected_status: InitialCheckStatus, is_leader: bool, hostname: str) -> None:
+        if not self.should_wait:
+            status_value, _ = self._read_state()
+            if status_value == InitialCheckStatus.SANITY_CHECK_FAILED.value:
+                pt_assert(False, "Leader node sanity check failed. Exiting the test.")
+
+            logger.info("Skip waiting and acknowledging status")
+            return
+
+        if wait_until(
+            432000,
```

Comment:
Based on the discussion with Jianquan, we can make the wait time to be very large or disable the wait here so we can have enough time to see what problems we will have in practice. This will make it easier for us to debug because the process will keep hanging there for very long. When we have a clear picture of what will happen during the wait time, we can change to a reasonable wait time here.

Therefore, setting the timeout for waiting and acknowledging status from leader to be 5 days (432000 seconds).